File: buysellstock.java
// TC=O(n)
// SC=o(1)
public class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;

        // Traverse through the prices array
        for (int i = 1; i < prices.length; i++) {
            // If the price on the current day is higher than the previous day, accumulate the profit
            if (prices[i] > prices[i - 1]) {
                maxProfit += prices[i] - prices[i - 1];
            }
        }

        return maxProfit;
    }

File: buy_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profits = 0

        for i in range(1, len(prices)):
            if prices[i] > prices[i - 1]:
                profits += prices[i] - prices[i - 1]
        return profits


# time complexity is O(n)
# space complexity is O(n)
